src/app/services/modal.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import { ModalWindowModel } from './model/modal.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export default class ModalWindowService {
+  private item: ModalWindowModel = { page: '', text: '', title: '', link: '' };
+
+  getData(value: string): ModalWindowModel {
+    switch (value) {
+      case 'Go to catalog':
```

Comment:
Let's create an enum with actions for switch case

Reply:
updared